Library/Homebrew/test/rubocops/cask/stanza_grouping_spec.rb

```diff
@@ -506,20 +506,76 @@
     include_examples "autocorrects source"
   end
 
-  # TODO: detect incorrectly grouped stanzas in nested expressions
-  context "when stanzas are nested in a conditional expression" do
-    let(:source) do
-      <<~CASK
-        cask 'foo' do
-          if true
-            version :latest
+  context "when stanzas are nested one-level in `on_*` blocks" do
+    describe "basic nesting" do
+      let(:source) do
+        <<~CASK
+          cask 'foo' do
+            on_arm do
+              version "1.0.2"
+
+              sha256 :no_check
+            end
+            on_intel do
+              version "0.9.8"
+              sha256 :no_check
+              url "https://foo.brew.sh/foo-intel.zip"
+            end
+          end
+        CASK
+      end
 
-            sha256 :no_check
+      let(:correct_source) do
+        <<~CASK
+          cask 'foo' do
+            on_arm do
+              version "1.0.2"
+              sha256 :no_check
+            end
+            on_intel do
+              version "0.9.8"
+              sha256 :no_check
+
+              url "https://foo.brew.sh/foo-intel.zip"
+            end
           end
-        end
-      CASK
+        CASK
+      end
+
+      include_examples "autocorrects source"
     end
 
-    include_examples "does not report any offenses"
+    # TODO: Maybe this should be fixed too?
```

Comment:
Is top-level `livecheck` inner grouping also handled by this cop or a separate one?

Reply:
Good question. We don't seem to do any detection of groups within `livecheck` blocks.

Reply:
I think this TODO can be removed, or changed.

I think we want a separate cop for the order inside of `livecheck` blocks. Unless it's easier to also implement it in this one?

Reply:
Bit late now, yeah, I'll remove it.